src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -204,6 +213,31 @@ async def do_update(self, job, data):
 
         return await self.config()
 
+    @private
+    async def destination_pool_error(self, new_pool):
+        config = await self.config()
+
+        try:
+            existing_dataset = await self.middleware.call('zfs.dataset.get_instance', config['basename'])
+        except InstanceNotFound:
+            return
+
+        used = existing_dataset['properties']['used']['parsed']
+
+        try:
+            new_dataset = await self.middleware.call('zfs.dataset.get_instance', new_pool)
+        except InstanceNotFound:
+            return f'Dataset {new_pool} does not exist'
+
+        available = new_dataset['properties']['available']['parsed']
+
+        used = int(used * 1.1)
```

Comment:
Why `1.1`?

Reply:
A safety margin because same files won't take exactly the same amount of space on a different pool.

Reply:
That's a great point, would you care to just add a small comment explaining why you're doing that?